src/main/gui/main-window.ts

```diff
@@ -281,10 +284,14 @@ const registerEventHandlerPostSetup = (
 
     const restartChainner = (): void => {
         if (backend.owned) {
-            backend.tryKill();
+            backend
+                .tryKill()
+                .then(() => {
+                    app.relaunch();
+                    app.exit();
+                })
```

Comment:
I would use `finally` here in case `tryKill` ever throws. I know that `tryKill` isn't supposed to throw, but we don't have a way to enforce this in the type system, so we can't trust it.

```suggestion
                .finally(() => {
                    app.relaunch();
                    app.exit();
                })
```